internal/pkg/cli/recorder/recorder.go

```diff
@@ -72,29 +78,46 @@ func (r *Recorder) Run() error {
 	}
 	defer r.UnloadBpfRecorder(r.bpfRecorder)
 
-	cmd := command.New(r.options.commandOptions)
-	pid, err := r.CommandRun(cmd)
-	if err != nil {
-		return fmt.Errorf("run command: %w", err)
-	}
+	var mntns uint32
+	if r.options.noStart {
+		// command execution is managed externally,
+		// so we play dumb and just wait for SIGINT.
+		ch := make(chan os.Signal, 1)
+		r.Notify(ch, os.Interrupt)
+		log.Print(WaitForSigIntMessage)
+		<-ch
```

Comment:
What if the process has already exited before setting up the notification, is this going to be stuck here forever?

Reply:
Thanks for clarification. If returns immediately when the process is existed, there isn't a problem.